src/OpenTelemetry.Extensions/TracerProviderBuilderExtensions.cs

```diff
@@ -53,14 +54,42 @@ public static class TracerProviderBuilderExtensions
     /// Adds the <see cref="BaggageActivityProcessor"/> to the <see cref="TracerProviderBuilder"/>.
     /// </summary>
     /// <param name="builder"><see cref="TracerProviderBuilder"/> to add the <see cref="BaggageActivityProcessor"/> to.</param>
+    /// <param name="baggageKeyPredicate">Predicate to determine which baggage keys should be added to the activity.</param>
     /// <returns>The instance of <see cref="TracerProviderBuilder"/> to chain the calls.</returns>
     public static TracerProviderBuilder AddBaggageActivityProcessor(
-        this TracerProviderBuilder builder)
+        this TracerProviderBuilder builder,
+        Predicate<string> baggageKeyPredicate)
     {
         Guard.ThrowIfNull(builder);
+        Guard.ThrowIfNull(baggageKeyPredicate);
+
+#pragma warning disable CA2000 // Dispose objects before losing scope
+        return builder.AddProcessor(new BaggageActivityProcessor(baggageKeyPredicate));
```

Comment:
Just a random idea how to solve @cijothomas's concerns about user code throwing.

```csharp
return builder.AddProcessor(sp => new BaggageActivityProcessor(
   key => 
   {
        try
        {
            return baggageKeyPredicate(key);
        }
        catch (Exception ex)
        {
            // TODO: Log to EventSource
            return false;
        }
   }));
```